kolena/_experimental/data_ingestion/_data_format/coco.py

```diff
@@ -0,0 +1,126 @@
+from kolena._experimental.data_ingestion._utils.io import load_json
+from kolena._experimental.object_detection import GroundTruth
+from kolena._experimental.object_detection import Model
+from kolena._experimental.object_detection import TestCase
+from kolena._experimental.object_detection import TestSample
+from kolena._experimental.object_detection import TestSuite
+from kolena.workflow import Model as BaseModel
+from kolena.workflow import TestCase as BaseTestCase
+from kolena.workflow import TestSuite as BaseTestSuite
+from kolena.workflow.annotation import LabeledBoundingBox
+
+
+class CocoJsonDataFormat(BaseDataFormat):
+    def __init__(self, config: DataIngestionConfig) -> None:
+        super().__init__(config)
+        self.raw_data = None
+
+    def load_data(self) -> None:
+        self.raw_data = load_json(self.config.data_paths[0])
+
+    def process_data(self) -> None:
+        self.image_map: Dict[int, Any] = self._get_image_map(self.raw_data)
+        self.label_map: Dict[int, str] = self._get_label_map(self.raw_data)
+        self.bbox_map = self._get_bbox_map(self.raw_data, self.image_map, self.label_map)
+
+    def save_data(self) -> None:
+        locator_prefix = self.config.locator_prefix
+        test_case_name = self.config.test_case_name
+        test_suite_name = self.config.test_suite_name
+        reset = self.config.reset
+
+        complete_test_case = self._create_complete_test_case(
+            self.bbox_map,
+            self.image_map,
+            reset,
+            test_case_name,
+            locator_prefix,
+        )
+        # create the test suite with the complete test case
+        _ = TestSuite(
+            test_suite_name,
+            description="data ingestion via Kolena package",
+            test_cases=[complete_test_case],
+            reset=reset,
+        )
+
+    def get_workflow(self) -> Tuple[Type[BaseTestCase], Type[BaseTestSuite], Type[BaseModel]]:
+        return TestCase, TestSuite, Model
+
+    def _get_image_map(self, raw_data):
+        return {int(record["id"]): record for record in raw_data["images"]}
+
+    def _get_label_map(self, raw_data):
+        return {int(record["id"]): record["name"] for record in raw_data["categories"]}
+
+    def _get_bbox_map(self, raw_data, image_map, label_map):
+        ids = set(image_map.keys())
+        image_to_boxes: Dict[int, List[LabeledBoundingBox]] = defaultdict(lambda: [])
+        for annotation in raw_data["annotations"]:
+            image_id = annotation["image_id"]
+            label = label_map[annotation["category_id"]]
+
+            # check that the box is in a valid image, not a crowd box
+            if image_id in ids and not annotation["iscrowd"]:
+                bbox = annotation["bbox"]
+                top_left = (bbox[0], bbox[1])
+                bottom_right = (bbox[0] + bbox[2], bbox[1] + bbox[3])
+                bounding_box = LabeledBoundingBox(top_left, bottom_right, label)
```

Comment:
COCO JSON also includes additional attributes inside bboxes (and can be extended with more attributes) — these should make their way into Kolena.